fsspec/implementations/local.py

```diff
@@ -95,20 +104,15 @@ def info(self, path, **kwargs):
                 t = "other"
         result = {
             "name": path,
-            "size": out.st_size,
+            "size": size,
             "type": t,
             "created": out.st_ctime,
             "islink": link,
         }
         for field in ["mode", "uid", "gid", "mtime", "ino", "nlink"]:
             result[field] = getattr(out, f"st_{field}")
-        if result["islink"]:
+        if link:
             result["destination"] = os.readlink(path)
```

Comment:
Since we are doing micro-optimisation, might the destination path also be known?

Reply:
I don't think there is a way to get `destination` path without doing a `readlink`. `stat` struct does not have a path or a target path.